subprojects/launcher/src/integTest/groovy/org/gradle/launcher/DeprecatedBuildIntegrationTest.groovy

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.launcher
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+import org.gradle.util.Requires
+
+@Requires(adhoc = { AvailableJavaHomes.getJdks("1.7") })
+class DeprecatedBuildIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
:x: Would be good to also have a test to check that running with Java 8 does not print the warning.

Reply:
Done.